ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/results/EngineUpdateForkChoiceResult.java

```diff
@@ -17,28 +17,27 @@
 import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
 
 import org.hyperledger.besu.consensus.merge.blockcreation.PayloadIdentifier;
-import org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.ExecutionEngineJsonRpcMethod.ForkChoiceStatus;
 
 import java.util.Optional;
 
 import com.fasterxml.jackson.annotation.JsonGetter;
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonPropertyOrder;
 
-@JsonPropertyOrder({"status", "payloadId"})
+@JsonPropertyOrder({"payloadStatus", "payloadId"})
 public class EngineUpdateForkChoiceResult {
-  private final ForkChoiceStatus status;
+  private final EnginePayloadStatusResult payloadStatus;
   private final PayloadIdentifier payloadId;
 
   public EngineUpdateForkChoiceResult(
-      final ForkChoiceStatus status, final PayloadIdentifier payloadId) {
-    this.status = status;
+      final EnginePayloadStatusResult payloadStatus, final PayloadIdentifier payloadId) {
+    this.payloadStatus = payloadStatus;
```

Comment:
As forkchoiceUpdated only allows a subset of `EnginePayloadStatusResult.status`, I think I would verify that nobody uses an invalid status here. I think throwing an exception if the status is `ACCEPTED` or `INVALID_BLOCKHASH` should be sufficient.

Reply:
I wasn't really keen on combining the two enums, but the execution spec specifically called out that both use the same payloadStatusV1 struct.  I can add some guard rails for those two invalid response status vals 👍